swap_meet/item.py

```diff
@@ -1,38 +1,28 @@
-
+from .constants import CONDITION_STATES
 import uuid
+import math
 
 class Item:
     def __init__(self, id=None, condition=0, age=None):
-        if id is not None and isinstance(id, int):
-            self.id = id
-        else:
-            self.id = uuid.uuid4().int  # long unique number
+        if id is not None and not isinstance(id, int):
+            raise TypeError("ID has to be an integer or not provided, so it can be safely set!")
+        self.id = id if id is not None else uuid.uuid4().int # long unique number  
 
-        if isinstance(condition, (int, float)) and 1.0 <= condition <= 5.0:
-            self.condition = condition
-        else:
-            self.condition = 0
+        if not isinstance(condition, (int, float)):
+            raise TypeError("Condition has to be a number between 0.0 and 5.0")
+        if not (0.0 <= condition <= 5.0):
+            raise ValueError("Condition has to be a number between 0.0 and 5.0")
+        self.condition = condition
             
         self.age = age
 
     def get_category(self):
-        return self.__class__.__name__
+        return "Item"
```

Comment:
👍 Nice to see that you swapped this over to an OOP implementation (with overridden subclass imeplementations) rather than relying on the introspection approach.